### DISCUSSION TITLE
Shiki js support

### DISCUSSION THREAD

**Discussion by c2tz** [imfing/hextra#187] [Ideas]
Hello, is support for Shiki (synthax hilighter which uses vscode's engine) planned for later? i had used a short code to replace chroma i don't think it's too useful but if needed i can share it. i would like to note that i was using with cdn which i think is not necessarily desirable.

**Comment by imfing** [comment]
Hi @c2tz thanks for the idea!

I'm absolutely open to include client-side code highlighter if it works better. If you have plan for making it a drop-in replacement for `chroma` in this theme, it would be great.

> I would like to note that i was using with cdn which i think is not necessarily desirable.

CDN is fine, but by default, we might need to ship the JS/CSS bundle like what we did in Hextra:
https://github.com/imfing/hextra/tree/main/assets/lib

**Reply by c2tz** [reply]
I'm just proposing the idea, I may not be competent enough to initialize the project here's the code I used (it was with papermod), I had to make some adjustments in the css and optimizations remains largely feasible. For the moment I don't know of any other hugo theme that uses Shiki for synthaxis (it's mainly hljs, prism or chroma) but if the aim of this theme is to get as close as possible to the spirit of Nextra (and in addition to having the best synthaxic coloring) here it is : 

https://github.com/shikijs/shiki/issues/495#issuecomment-1646632297

And the rendering of this code: 

![280416311-8203eb65-a863-4726-82d0-e267832e1d32](https://github.com/imfing/hextra/assets/95414249/5ee6a721-1cbd-47fb-831b-ee73e58e696d)

As I said, I'm just opening up the idea, but if someone more competent or I ever manage to implement it or even you run out of ideas one day and want to integrate shiki, at least you'll have some content.

this must be taken into consideration :

https://github.com/antfu/shikiji

**Reply by imfing** [reply]
Thanks! I'll keep this in mind. If I have the time, I'll try to add it to Hextra 😃. It should be pretty straightforward, similar to how Mermaid is included.

---

>  if the aim of this theme is to get as close as possible to the spirit of Nextra (and in addition to having the best synthaxic coloring)

`get as close as possible to the spirit of Nextra` was never the goal of this theme.
Instead, it was designed for providing a streamlined and enhanced experience for creating static websites, whether for documentation, blogs, or portfolios. Think of it as the spiritual successor to Nextra, tailored specifically for Hugo, which is why I named it Hextra.
